File: eshop-integration/eAPI/v1/java/eAPI-v1-connector/src/main/java/cz/monetplus/mips/eapi/v1/connector/entity/PayInitReq.java
package cz.monetplus.mips.eapi.v1.connector.entity;

import java.util.List;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;

@XmlRootElement
@XmlAccessorType(XmlAccessType.FIELD)
public class PayInitReq extends SignBase {

	private static final long serialVersionUID = -438284924457338635L;

	@XmlElement(nillable=false)
	public String merchantId;
	
	@XmlElement(nillable=false)
	public String orderNo;

	@XmlElement(nillable=false)
	public String payOperation;
	
	@XmlElement(nillable=false)
	public String payMethod;

	@XmlElement(nillable=false)
	public long totalAmount;

	@XmlElement(nillable=false)
	public String currency;

	@XmlElement(nillable=false)
	public boolean closePayment;
	
	@XmlElement(nillable=false)
	public String returnUrl;

	@XmlElement(nillable=false)
	public String returnMethod;

	@XmlElement(nillable=false)
	public List<CartItem> cart;
	
	@XmlElement(nillable=false)
	public String description;

	@XmlElement(nillable=true)
	public String merchantData;

	@XmlElement(nillable=true)
	public String customerId;
	
	@XmlElement(nillable=true)
	public String language;
	
	@Override
	public String toSign() {
		StringBuffer sb = new StringBuffer();
		add(sb, merchantId);
		add(sb, orderNo);
		add(sb, dttm);
		add(sb, payOperation);
		add(sb, payMethod);
		add(sb, totalAmount);
		add(sb, currency);
		add(sb, closePayment);
		add(sb, returnUrl);
		add(sb, returnMethod);
		for (CartItem item : cart) {
			add(sb, item.toSign());
		}
		add(sb, description);
		add(sb, merchantData);
		add(sb, customerId);
		add(sb, language);
		deleteLast(sb);
		return sb.toString();
	}

}

File: Integration Examples/eAPI v1.6/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/PayInitReq.cs
using System.Collections.Generic;
using System.ComponentModel;
using System.Text;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects
{
    public class PayInitReq : SignBaseRequest
    {
        [JsonProperty("orderNo")]
        public string OrderNo { get; set; }

        [JsonProperty("payOperation")]
        public string PayOperation { get; set; }

        [JsonProperty("payMethod")]
        public string PayMethod { get; set; }

        [JsonProperty("totalAmount")]
        public long TotalAmount { get; set; }

        [JsonProperty("currency")]
        public string Currency { get; set; }

        [JsonProperty("closePayment")]
        public bool ClosePayment { get; set; }

        [JsonProperty("returnUrl")]
        public string ReturnUrl { get; set; }

        [JsonProperty("returnMethod")]
        public string ReturnMethod { get; set; }

        [JsonProperty("cart")]
        public List<CartItem> Cart { get; set; }

        [JsonProperty("description")]
        public string Description { get; set; }

        [JsonProperty("merchantData")]
        public string MerchantData { get; set; }

        [JsonProperty("customerId")]
        public string CustomerId { get; set; }

        [JsonProperty("language")]
        public string Language { get; set; }

        [JsonProperty("ttlSec")]
        public int TtlSec { get; set; }

        [JsonProperty("logoVersion")]
        public int LogoVersion { get; set; }

        [JsonProperty("colorSchemeVersion")]
        public int ColorSchemeVersion { get; set; }

        public override string ToSign()
        {
            StringBuilder sb = new StringBuilder();
            Add(sb, MerchantId);
            Add(sb, OrderNo);
            Add(sb, DateTime);
            Add(sb, PayOperation);
            Add(sb, PayMethod);
            Add(sb, TotalAmount);
            Add(sb, Currency);
            Add(sb, ClosePayment.ToString().ToLower());
            Add(sb, ReturnUrl);
            Add(sb, ReturnMethod);

            foreach (var item in Cart)
                Add(sb, item.ToSign());

            Add(sb, Description);
            Add(sb, MerchantData);
            Add(sb, CustomerId);
            Add(sb, Language);
            Add(sb, TtlSec);
            Add(sb, LogoVersion);
            Add(sb, ColorSchemeVersion);
            DeleteLast(sb);
            return sb.ToString();
        }
    }
}
